# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Iteration.cpp
# Commit: c71e5fccc8e6
# Full Hash: c71e5fccc8e6b2816158a1afb531a510105f174f
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-09-22 04:09:13
# Description:
#   Bug 1731540: Handle empty iterators when unwinding uncatchable exceptions r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D126125
# ==============================================================================

diff -r 000302929625 -r c71e5fccc8e6 js/src/vm/Iteration.cpp
--- a/js/src/vm/Iteration.cpp	Tue Sep 21 16:38:54 2021 +0000
+++ b/js/src/vm/Iteration.cpp	Tue Sep 21 17:01:33 2021 +0000
@@ -1383,6 +1383,9 @@
 void js::UnwindIteratorForUncatchableException(JSObject* obj) {
   if (obj->is<PropertyIteratorObject>()) {
     NativeIterator* ni = obj->as<PropertyIteratorObject>().getNativeIterator();
+    if (ni->isEmptyIteratorSingleton()) {
+      return;
+    }
     ni->unlink();
   }
 }
